docs/buildctl.md

```diff
@@ -0,0 +1,195 @@
+```
+
+`buildctl build` uses a buildkit daemon `buildkitd` to drive a build.
+
+The build consists of the following key elements:
+
+* [frontend definition](#frontend): parses the build descriptor, e.g. dockerfile
+* [local sources](#local_sources): sets relevant directories and files passed to the build
+* [frontend options](#frontend_options): options that are relevant to the particular frontend
+* [output](#output): defines what format of output to use and where to place it
+* [cache](#cache): defines where to export the cache generated during the build to, or where to import from
+
+### frontend
+
+The frontend is declared by the flag `--frontend <frontend>`. The `<frontend>` must be one built into `buildkitd`, or an OCI
+image that implements the frontend API.
+
+In the above example, we are using the built-in `dockerfile.v0` frontend, which knows how to parse a dockerfile and convert it to LLB.
+
+### local sources
+
+A build may need access to various sources local to the `buildctl` execution environment,
+such as files and directories or OCI images. These can be provided from the local
+environment to which the user of `buildctl` has access. These are provided as:
+
+* `--local <name>=<dir>` - allow buildkitd to access a local-to-buildctl directory `<dir>` under the unique name `<name>`.
+* `--oci-layout <name>=<dir>` - allow buildkitd to access OCI images in the local-to-buildctl directory `<dir>` under the unique name `<name>`.
+
+Each of the above is expected to provide a unique name, for this invocation of `buildctl`, for a directory. Other parts of `buildctl` can then
+use those "named contexts" to reference directories, files or OCI images.
+
+For example:
+
+```
+buildctl build --local test1=/var/lib/test1
+```
+
+lets `buildkitd` access all of the files in `/var/lib/test1` (relative to wherever `buildctl` is running), referenced via the name `test1`.
+
+Similarly:
+
+```
+buildctl build --oci-layout foo=/var/lib/oci
+```
+
+lets `buildkitd` access OCI images under `/var/lib/oci` (relative to wherever `buildctl` is running), referenced via the name `test1`.
+
+These "named references" are used by the frontend, either directly or with explicit options.
+
+#### dockerfile frontend sources
+
+The dockerfile frontend, enabled via `buildctl build --frontend=dockerfile.v0`, expects to have access to 2 named references:
+
+* `context`: where to perform the build.
+* `dockerfile`: where to find the dockerfile to parse describing the build.
+
+Thus, a dockerfile build invocation would include:
+
+```
+buildctl build --frontend dockerfile.v0 --local context=. --local dockerfile=.
+```
+
+The above means, "build using the dockerfile frontend, passing it the context of the current directory where I am running `buildctl`, and the
+dockerfile in the current directory as well."
+
+### frontend options
+
+Frontend-specific options are defined via `--opt <key>=<value>`. The specific meanings of those are frontend-speciifc.
```

Comment:
```suggestion
Frontend-specific options are defined via `--opt <key>=<value>`. The specific meanings of those are frontend-specific.
```

Reply:
It is so specific, that it writes specific as speciific, in a way no one else does. 😆 

I will fix it.

Reply:
Fixed